### PR TITLE
[stdlib] Uninline Existential Collections

### PR DESCRIPTION
**Pull request by airspeedswift**
Similar to #17476 but for `AnySequence` and friends. Many of these should be unnecessary since they're internal types/methods. Removing them all might hurt performance but lets start with this and add some back in where appropriate.

### CODE REVIEW

**Comment by airspeedswift** [comment]
@swift-ci please smoke benchmark

File: stdlib/public/core/ExistentialCollection.swift.gyb [hunk 1]
```diff
@@ -672,22 +570,19 @@ extension _ClosureBasedSequence: Sequence {
 /// An instance of `AnySequence` forwards its operations to an underlying base
 /// sequence having the same `Element` type, hiding the specifics of the
 /// underlying sequence.
-//@usableFromInline
+//
 @_fixed_layout
 public struct AnySequence<Element> {
```
Comment (slavapestov) [inline stdlib/public/core/ExistentialCollection.swift.gyb]: I'd be curious about the benchmark results for un-fixed-layout'ing these, in a separate PR

**Comment by airspeedswift** [comment]
@swift-ci please smoke test compiler performance

**Comment by airspeedswift** [comment]
@swift-ci please test

**Comment by airspeedswift** [comment]
lol

**Comment by airspeedswift** [comment]
@swift-ci please smoke benchmark

**Pull request closed without merging**